README.md

```diff
@@ -1,2 +1,128 @@
-# zarr_linked_data
-The project seeks to make a dataflow composed both of the Zarr data format and linked metadata.
+# Zarr Linked Data
+
+The project seeks to make **a dataflow composed both of the Zarr data format and linked metadata (JSON-LD)**.
+
+The current **infrastructure prototype** implementation is kubernetes based, requiring Metaflow for code pipelines, MinIO for storage and Argo for automation. 
+
+The project is a prototype infrastructure **under construction** and is still incomplete. Its components need to be assessed and adapted to a new use-case before being used.  
+
+<img width="919" alt="Screenshot 2023-12-13 at 15 04 29" src="https://github.com/SDSC-ORD/zarr-linked-data/assets/22447169/ed7153f0-e22b-46a9-bf1f-fffb3a7f5445">
+
+The following sections describe the project further: 
+- A. Project Description
+- B. Getting Started 
+- C. Example / Usage 
+
+# A. Project Description
+
+The project's main interesting features are pipelines to manipulate [Zarr data formats](https://zarr.dev). Zarr files can store both data and [metadata](https://zarr.readthedocs.io/en/stable/tutorial.html#user-attributes), in a [hierarchical manner](https://zarr.readthedocs.io/en/stable/tutorial.html#groups). Zarr is best for storing any array-like data, as it allows compression and chunking. The metadata present with the data makes the Zarr file format of high interest for interoperability and findability goals of [FAIR principles](https://www.go-fair.org/fair-principles/). 
+
+This project came into being tied to the [Cat+ initiative from EPF domain in Switzerland](https://swisscatplus.ch). From a fully automated chemistry lab, numerous samples will be processed under different parameters by a variety of machines. The idea is to have the output data of this lab, and even more so the associated metadata, to be queryable and retrievable. On the long run, the idea is also to allow external sources to contribute their data to the system as well, following metadata standards. 
+
+The *main functionalities* that were implemented in this project were: 
+- Retrieve (["consolidate"](https://zarr.readthedocs.io/en/stable/tutorial.html#consolidating-metadata) in zarr jargon) all the metadata from the numerous arrays of data
+- From a metadata Universal Resource Identifier [URI](https://www.w3.org/wiki/URI), retrieve the associated dataset in the [Zarr store](https://zarr.readthedocs.io/en/stable/api/storage.html)
+
+# B. Getting Started
+
+## Locally (without k8)
+
+Set-up you [poetry project](https://python-poetry.org/docs/#installation) using `poetry install` in the same folder as the pyproject.toml
+
+OR 
+
+Install the requirements with `pip install -r requirements.txt` (these are the requirements for the kubernetes set-up so you will install packages which are not needed for local dev)
```

Comment:
**question:** Is there a specific reason to keep  k8s deps in the requirements.txt, instead of making a dependency group in the pyproject.toml?

This would also reduce the duplication, with all core pkgs being defined once, so users can then do `poetry install --with k8s`

Reply:
On the cat+ server side, I couldn't install poetry (did not have the rights) so I had to work with requirements.txt 
Hence the current set-up ! :) So I would prefer to keep it this way. In the future if the prototype on Cat+ server gets dismantled, then I will bifurk the project entirely to poetry